app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails5_api_controller_backport'
+
+module Api
+  module V1
+    class BaseController < ActionController::API
+      private
+
+      def authenticate
```

Comment:
We might need to add another layer of security, such as API key sent in Authorization headers. It can be static, defined in environment, and not backed by an actual registry user.

Reply:
Authentication based on IP _only_ is pretty weak, they only validate the host, and not the actual user. For more info:

https://security.stackexchange.com/questions/4533/in-what-scenarios-is-relying-on-source-ip-address-as-a-security-control-acceptab

https://security.stackexchange.com/questions/13255/how-secure-can-ip-based-login-be

Reply:
I don't argue that it's weak. My point was to keep it as is for the launch, given it is not accessible from Internet.

Perhaps we will need some message broker instead of API, such as rabbitmq...